### DISCUSSION TITLE
Github/wiki/thread

### DISCUSSION THREAD

**Discussion by likeproblem** [likeproblem/RestockRecolour#11] [Collaborator Disscusion]
VERY_COOL_PLACEHOLDER

**Comment by likeproblem** [comment]
@UncleMateo I won't be updating wiki until it is properly implemented from your side, is that okay?

**Reply by UncleMateo** [reply]
I will try and get the rest of the Wiki-Parts and the Key for the wiki completed within the next few hours, and then I plan on a combination of experimenting with embedding images in the wiki (to figure out how to handle the images of the parts, what looks clean and informative, what doesn't), and working on my next cluster of recolors for the cargo/inventory parts (there's only 4 in the whole category)

**Reply by likeproblem** [reply]
Also I think of maybe another categories list/ another wiki page solely for pictures? Because if we put them into the wiki, it will bloat it very quick

**Reply by likeproblem** [reply]
I see the wiki should be updated by now, I will just go over and make sure every part is included

**Reply by UncleMateo** [reply]
You know how each part category tab can be minimized? Can we shove a sub-tab into that? Or is that not how it works?
  I have some other ideas that might work, making standardized 512x512 scaled cropped screenshots, putting them at the bottom of a given parts tab.  Just an idea, I didn't have time to test it.  Also might look like crap, which is why I wanted to play around with the 'Preview' thing a bit, see if there's any way to do either of the above.. but another idea could be to populate that empty front page of the wiki with a screenshot gallery?

**Reply by likeproblem** [reply]
Ah yes, home page would be great! And yes, you can make sub categories in categories, I will make placeholders for that

**Reply by likeproblem** [reply]
I meant I will make placeholders for albums on home page, pardon me

**Reply by likeproblem** [reply]
@UncleMateo I made placeholder album page in Home

**Comment by likeproblem** [comment]
@UncleMateo check out my update, I checked of several items and *tried* to count them, all should be checked off just fine but please check per-section counters

**Reply by UncleMateo** [reply]
Thank you for doing that! I will also double check there aren't any missing parts with recolors, and then also make sure my categorization of parts is correctly labeled, I think I may have had more than one symbol for "Shouldn't be affected by Restock but for some reason its TURD recolor is broken/ just red" and I need to pull up KSP to double check those as well.  They are marked by both '❓' and '‼‼' right now, I believe.  I became unexpectedly busy today, but I'll try and fix/polish/update it as soon as I can.

**Reply by UncleMateo** [reply]
Per section counters were all kinds of screwed up, some were off by 1, some by 10, but they're fixed now.

**Comment by likeproblem** [comment]
@UncleMateo I submitted my commit, and seems 5m tanks are missing, perhaps you forgot some dlc parts?

**Reply by UncleMateo** [reply]
I did, should be corrected now.

**Comment by likeproblem** [comment]
@UncleMateo I have an idea for wiki. There should be Quality Check, I will carefully go over each part, and if they okay, mark with ✴️ in wiki, otherwise fix them. Also 🔄 for if variants are doable and should be done.

**Reply by UncleMateo** [reply]
GREAT idea.. I also think if we marked parts that have some variant issue, either a mesh that won't recolor, or a variant that doesn't recolor.  You know, a part we added which has recolor but has some kind of issue with it.

**Reply by likeproblem** [reply]
Oh a problem can be marked with ❗

**Comment by likeproblem** [comment]
@Spartwo I made an usefull utility for wiki! You just paste in wiki text, and it will tell what is missing(curently cupola and 5m tanks). I also work on to add another function which will tell what parts are not checked off!

Bunch of these changes should be available in ~4 hours, a lot of work still to do

**Reply by likeproblem** [reply]
Oh and obviosly tell all parts present

**Comment by likeproblem** [comment]
@UncleMateo @Spartwo very important 2 commits, please make sure you backup your changes and reapply them

https://github.com/likeproblem/RestockRecolour/commit/96442c926c78e385695e12eab96ed7dce4c75d34 - the hazardousz commit
https://github.com/likeproblem/RestockRecolour/commit/0f7babcd935bf3fefbc52c5d3e725cc652f77548 - my normal one

Very fun changes too, a lot of them

**Reply by UncleMateo** [reply]
My additions have been tested and seem to work, noticed the filepath change and made corrections.

**Reply by likeproblem** [reply]
> My additions have been tested and seem to work, noticed the filepath change and made corrections.

Oh I forgot to apply some path changes to ConfigMaker, will do that!

**Comment by likeproblem** [comment]
@UncleMateo gonna make some pics :P, and I made two cfgs to blacklist or whitelist colored parts

**Reply by UncleMateo** [reply]
Nice!  Just wanted to report that somehow a dozen parts were mislabeled as Restock, inflating the count.  When I fixed it, we went up a full percentage point.  Long story short, I'm one of those early early access people who got the expansions for free, so I get confused as to what parts are x-pack related.  Well, when I went to get started on the multi-sized square and triangle panels, I realized they were MakingHistory parts, they had no Restock patches, and their recolors were just broken.  So I fixed the wiki and set those parts aside.  Maybe when we're in the region of like 70-80% done I might start putting a small amount of my time into figuring out how to make compatibility patches for those many parts.  Seems like just about all the landing gears and these panels are the most egregious offenders, but there's quite a few that should be recolorable but just aren't for whatever reason.

**Reply by likeproblem** [reply]
I have idea how to fix recolours. If they don't have any model changes, we can just make our own recolours BUT cross reference masks from TURD

**Reply by Spartwo** [reply]
Making history panels don't work because they're falsely including the !restock requirement

I've had a branch of TU_MHBG for months because of an unrelated engine plate issue. So the options are either to commit fully to a branched version and get manwithnoname to make the swapover or to include config duplicates.

I say the latter, people are gonna get stuck using the wrong version anyway because they won't update

**Reply by likeproblem** [reply]
@Spartwo thanks for doing that! So the plan is just copy paste configs? I recommend doing them in separate folder/configs, like name them TU_Restock_Default_Broken

**Reply by Spartwo** [reply]
> @Spartwo thanks for doing that! So the plan is just copy paste configs? I recommend doing them in separate folder/configs, like name them TU_Restock_Default_Broken

Exactly that TU_MHBG_Fix

**Reply by likeproblem** [reply]
Oh that's a better solution to a problem, like an optional patch nearby TU_Restock

**Reply by Spartwo** [reply]
I'm including a pylons patch for now until the forums are back and the error can be noted

**Comment by likeproblem** [comment]
@UncleMateo I think quality control should be only applied to parts which have all variants, because quality control mark should show "part is fully done, ignore"

**Reply by Spartwo** [reply]
Agreed, eventual goal should not be to have variants unpainted.

**Reply by UncleMateo** [reply]
Okay that makes a lot of sense.  Forgive my misunderstanding.  I'll change the definition in the wiki a bit to make it consistent with that ideal.

**Comment by EmberSkyMedia** [comment]
Its is possible for me to help with the process? (PicoSpace on KSP forums / discords).
I'm not the best at configs but I'm pretty talented with artwork and can do meshes as my understanding is we are taking the orignal meshes and apply RGB versions (layers/paint) an overlay so that when appling paint jobs in TU.

I just don't want to step on people's toes.

**Reply by likeproblem** [reply]
Hello, nice to see you here!

So the thing is this, check Guidelines first and ingame examples too, then fork and use git/Github Desktop to copy. Then do any changes you want and commit+push, after you are done with stuff you can pull request.

You have discord? I can explain the rest of the way there because I don't know how to explain it in orderly manner here(Im bad at communicating sorry)

**Reply by UncleMateo** [reply]
Hey, welcome!  So far we've been coordinated who's doing what part off of the 'part association list' on the wiki, not sure if just anybody can edit that, though, it may be just us 3 for the moment, so I think other collaborators should probably use the discussion board topic 'Part Plans' (https://github.com/likeproblem/RestockRecolour/discussions/8)

I'll try to check both places from now on so that I don't accidentally duplicate someone else's efforts. @likeproblem @Spartwo what do you think of that idea?

**Reply by likeproblem** [reply]
That sounds great!

**Comment by likeproblem** [comment]
@Spartwo @UncleMateo I created https://github.com/likeproblem/RestockRecolour/wiki/Advanced-Tactics, drop off any useful config tricks!

**Reply by UncleMateo** [reply]
Oooh will do.  I have the day off tomorrow, depending on how tricky the parts I have in queue right now end up being, I think I'll plug in a bit of time trying to add some stuff there.
